cli/src/main.rs

```diff
@@ -41,44 +42,102 @@ struct Args {
+    /// Store the given WASM task type in the mesh.
     #[clap(display_order = 1)]
+    Store {
+        /// Path to the task manifest file.
+        manifest: PathBuf,
+    },
+    /// Run the specified WASM binary.
+    #[clap(display_order = 2)]
     Run {
-        /// A descriptive name for the job
-        #[clap(long, short)]
-        name: Option<String>,
-        /// A description for the job
-        #[clap(long, short)]
-        description: Option<String>,
-        /// The file containing the wasm binary to run
-        #[clap(value_name = "WASM BINARY")]
-        binary_file: PathBuf,
+        /// The name of the previously-stored job to run.
+        name: String,
         /// Path to a file to pass to the binary; omit to read from stdin (if present)
-        #[clap(value_name = "OPTIONAL INPUT TO WASM BINARY")]
         input_file: Option<PathBuf>,
-        /// Path to write the output of the job. Omit to write to stdout.
+        /// Path to write the output of the job; omit to write to stdout
         output_file: Option<PathBuf>,
     },
     /// Get the status of a job in progress.
-    #[clap(display_order = 2)]
+    #[clap(display_order = 3)]
     Status { id: Uuid },
     /// Get results for a job run, given its ID.
-    #[clap(display_order = 3)]
+    #[clap(display_order = 4)]
     Results { id: Uuid },
     /// Get full job run history from the running process.
-    #[clap(display_order = 4)]
+    #[clap(display_order = 5)]
     History,
+    /// Liveness check: ping at least one node on the mesh.
+    Ping,
 }
 
 static SERVAL_NODE_URL: Mutex<Option<String>> = Mutex::new(None);
 
 /// Convenience function to build urls repeatably.
-fn build_url(path: String) -> String {
+fn build_url(path: String, version: Option<&str>) -> String {
     let baseurl = SERVAL_NODE_URL.lock().unwrap();
     let baseurl = baseurl
         .as_ref()
         .expect("build_url called while SERVAL_NODE_URL is None");
-    format!("{baseurl}/v1/{path}")
+    if let Some(v) = version {
+        format!("{baseurl}/v{v}/{path}")
+    } else {
+        format!("{baseurl}/{path}")
+    }
+}
+
+fn upload_manifest(manifest_path: PathBuf) -> Result<()> {
+    println!("Reading manifest: {}", manifest_path.display());
+    let manifest = Manifest::from_file(&manifest_path)?;
+
+    println!("Reading WASM executable:{}", manifest.binary().display());
+    let executable = read_file(manifest.binary().to_path_buf())?;
+
+    let client = reqwest::blocking::Client::builder()
+        .timeout(Duration::from_secs(60))
+        .build()?;
+
+    // Start building pretty output now that we're past the most likely errors.
+    println!();
+    let mut table = Table::new();
+    table.set_format(*prettytable::format::consts::FORMAT_CLEAN);
+    table.add_row(row!["WASM task name:", manifest.fq_name()]);
+    table.add_row(row!["Version:", manifest.version()]);
+
+    let url = build_url("storage/manifests".to_string(), Some("1"));
```

Comment:
We pass `Some("1")` around often enough that a `build_url_v1` helper function probably wouldn't be a bad idea at some point.